### PR TITLE
Add steps to configure standalone Hammer on EL 9

### PR DESCRIPTION
**Pull request by asteflova**
#### What changes are you introducing?

Adding a procedure to configure standalone Hammer on EL 9.

#### Why are you introducing these changes? (Explanation, links to references, issues, etc.)

https://issues.redhat.com/browse/SAT-27900

#### Anything else to add? (Considerations, potential downsides, alternative solutions you have explored, etc.)

I used https://github.com/theforeman/foreman-documentation/blob/master/guides/common/modules/proc_preparing-a-host-for-external-databases.adoc as a template for how to structure a procedure for EL8 /EL 9.

#### Checklists

* [x] I am okay with my commits getting squashed when you merge this PR.
* [x] I am familiar with the [contributing](https://github.com/theforeman/foreman-documentation/blob/master/CONTRIBUTING.md) guidelines.

Please cherry-pick my commits into:

* [x] Foreman 3.12/Katello 4.14 (Satellite 6.16)
* [ ] Foreman 3.11/Katello 4.13
* [ ] Foreman 3.10/Katello 4.12
* [ ] Foreman 3.9/Katello 4.11 (Satellite 6.15; orcharhino 6.8/6.9/6.10)
* [ ] Foreman 3.8/Katello 4.10
* [ ] Foreman 3.7/Katello 4.9 (Satellite 6.14)
* [ ] Foreman 3.6/Katello 4.8
* [ ] Foreman 3.5/Katello 4.7 (Satellite 6.13; orcharhino 6.6/6.7)
* We do not accept PRs for Foreman older than 3.5.

### CODE REVIEW

File: guides/common/modules/proc_installing-standalone-hammer.adoc [hunk 1]
```diff
@@ -3,11 +3,34 @@
 
 You can install Hammer on a host running {install-on-os} that has no {ProjectServer} installed, and use it to connect from the host to a remote {Project}.
 
+.Prerequisites
 .Prerequisites
```
Comment (ofedoren) [inline guides/common/modules/proc_installing-standalone-hammer.adoc:7]: Does this mean a duplicate?

**Review by ofedoren — APPROVED** [review]
LGTM, just a small question:

File: guides/common/modules/proc_installing-standalone-hammer.adoc (hunk 1 above)
Comment (asteflova) [reply guides/common/modules/proc_installing-standalone-hammer.adoc]: It looks like it although it isn't showing in the preview. In any case, good catch, I'm dropping the extra line.

**Comment by asteflova** [comment]
> LGTM, just a small question:

Thanks!

Dear writers, this PR is now ready for a style review.

File: guides/common/modules/proc_installing-standalone-hammer.adoc [hunk 2]
```diff
@@ -7,7 +7,29 @@ You can install Hammer on a host running {install-on-os} that has no {ProjectSer
...
+ifdef::katello,orcharhino[]
+** {EL} 9 BaseOS
+** {EL} 9 AppStream
+endif::[]
+ifdef::katello[]
+** https://yum.theforeman.org/releases/{ProjectVersion}/el8/x86_64/foreman-release.rpm[{Project} release RPM]
+endif::[]
+
++
+For more information, see {ContentManagementDocURL}Synchronizing_Repositories_content-management[Synchronizing Repositories] in _{ContentManagementDocTitle}_.
```
Comment (maximiliankolb) [inline guides/common/modules/proc_installing-standalone-hammer.adoc:28]:
```suggestion
For more information, see {ContentManagementDocURL}Synchronizing_Repositories_content-management[Synchronizing repositories] in _{ContentManagementDocTitle}_.
```

**Review by maximiliankolb — APPROVED** [review]
one tiny suggestion, rest LGTM.

File: guides/common/modules/proc_installing-standalone-hammer.adoc [hunk 3]
```diff
@@ -7,7 +7,29 @@ You can install Hammer on a host running {install-on-os} that has no {ProjectSer
...
+** {RepoRHEL9BaseOS}
+** {RepoRHEL9AppStream}
+** {RepoRHEL9ServerSatelliteUtils}
+endif::[]
+ifdef::katello,orcharhino[]
+** {EL} 9 BaseOS
+** {EL} 9 AppStream
+endif::[]
+ifdef::katello[]
+** https://yum.theforeman.org/releases/{ProjectVersion}/el8/x86_64/foreman-release.rpm[{Project} release RPM]
```
Comment (ekohl) [inline guides/common/modules/proc_installing-standalone-hammer.adoc:24]:
```suggestion
** https://yum.theforeman.org/releases/{ProjectVersion}/el9/x86_64/foreman-release.rpm[{Project} release RPM]
```

File: guides/common/modules/proc_installing-standalone-hammer.adoc [hunk 4]
```diff
@@ -23,11 +45,62 @@ endif::[]
...
+For more information, see {ManagingHostsDocURL}Enabling_Custom_Repositories_on_Content_Hosts_managing-hosts[Enabling repositories on hosts] in _{ManagingHostsDocTitle}_.
+endif::[]
+ifdef::foreman-el[]
++
+[options="nowrap" subs="verbatim,quotes,attributes"]
+----
+# {client-package-install-el9} https://yum.theforeman.org/releases/{ProjectVersion}/el9/x86_64/foreman-release.rpm
+----
+endif::[]
+ifdef::foreman-deb[]
```
Comment (ekohl) [inline guides/common/modules/proc_installing-standalone-hammer.adoc:74]: Why are there Debian blocks under the EL 9 header? I get the impression the procedures share so little that it may be easier to write out all variants instead of merging them with many ifdefs in them.

**Review by ekohl — CHANGES_REQUESTED** [review]

File: guides/common/modules/proc_installing-standalone-hammer.adoc (hunk 4 above)
Comment (asteflova) [reply guides/common/modules/proc_installing-standalone-hammer.adoc:74]: Writing the procedures out is a good idea, that's actually how I got things this mixed up -- I must have gotten lost in the ifdefs. I'll see what I can come up with. Thanks!

Comment (maximiliankolb) [reply guides/common/modules/proc_installing-standalone-hammer.adoc:74]: I am also guilty as a reviewer: I made sure that every line was correct; ifdefs matched endifs; but did not see the bigger picture. :see_no_evil:

**Comment by asteflova** [comment]
Before, I tried to mimic the structure used in EL9/EL8 procedures [elsewhere](https://github.com/theforeman/foreman-documentation/blob/master/guides/common/modules/proc_preparing-a-host-for-external-databases.adoc). After experimenting a bit more with this particular module, I don't think that established structure would work here. For the standalone Hammer module, the difference between EL9/EL8 is very minor and I think that inserting one "If..." step works better. So that's what I went with.

In its current state, this PR reviews how ifdefs are used in the module and implements reuse only when larger, meaningful parts of the Prerequisites or Procedure are identical. When only a single step or a part of a bulleted list is identical, I chose to duplicate it rather than use an ifdef.

TL;DR: @maximiliankolb @ekohl Can you please re-review?

File: guides/common/modules/proc_installing-standalone-hammer.adoc [hunk 5]
```diff
@@ -4,74 +4,124 @@
...
+** {RepoRHEL8ServerSatelliteUtils}
 endif::[]
 
 .Procedure
+ifdef::foreman-deb[]
 . Enable the required repositories on the host.
-ifdef::katello,orcharhino,satellite[]
-For more information, see {ManagingHostsDocURL}Enabling_Custom_Repositories_on_Content_Hosts_managing-hosts[Enabling repositories on hosts] in _{ManagingHostsDocTitle}_.
++
+:distribution-codename: My_Distribution_Codename
```
Comment (maximiliankolb) [inline guides/common/modules/proc_installing-standalone-hammer.adoc:46]: This could be a problem to overwrite the attribute. Could you add "parent-distribution-codename: {distribution-codename}" and then reset+unset it afterwards?

File: guides/common/modules/proc_installing-standalone-hammer.adoc [hunk 6]
```diff
@@ -4,74 +4,124 @@
...
 endif::[]
+ifdef::katello,orcharhino,satellite[]
+. Enable the required repositories on the host.
+. If you are installing on {EL}{nbsp}8, enable the following module:
++
+[options="nowrap" subs="verbatim,quotes,attributes"]
+----
+# dnf module enable {dnf-module-utils}
+----
+. Install Hammer CLI.
```
Comment (maximiliankolb) [inline guides/common/modules/proc_installing-standalone-hammer.adoc:108]:
```suggestion
. Install Hammer CLI:
```

File: guides/common/modules/proc_installing-standalone-hammer.adoc [hunk 7]
```diff
@@ -4,74 +4,124 @@
...
+----
+. Set the `:host:` entry in the `/etc/hammer/cli.modules.d/foreman.yml` file to the {Project} URL:
++
+[options="nowrap", subs="+quotes,attributes"]
+----
+:host: 'https://_{foreman-example-com}_'
+----
 endif::[]
+ifdef::katello,orcharhino,satellite[]
+. Enable the required repositories on the host.
```
Comment (maximiliankolb) [inline guides/common/modules/proc_installing-standalone-hammer.adoc:101]: I would prefer an unordered substep here in favor of the list of prerequisites. Your call though.

File: guides/common/modules/proc_installing-standalone-hammer.adoc (hunk 5 above)
Comment (asteflova) [reply guides/common/modules/proc_installing-standalone-hammer.adoc]: How about I just drop the attribute to simplify things and use hard-coded `_My_Distribution_Codename_` in the command? I don't see any purpose in using the attribute actually, it was already there when I started updating the module. Or am I missing something @maximiliankolb?

File: guides/common/modules/proc_installing-standalone-hammer.adoc (hunk 6 above)
Comment (asteflova) [reply guides/common/modules/proc_installing-standalone-hammer.adoc]: Fixed.

File: guides/common/modules/proc_installing-standalone-hammer.adoc (hunk 7 above)
Comment (asteflova) [reply guides/common/modules/proc_installing-standalone-hammer.adoc:100]: I don't think so..? The prerequisite talks about enabling and synchronizing the repositories on the server. This step talks about enabling the repositories for the host on which you are installing Hammer. Those are two different actions.

Now I thought about moving this step to the prerequisites section to keep both actions together but then again, that would make the foreman-deb procedure look very different (because the step for foreman-deb includes a command block).

So, no change, but please, do let me know if you think something about my reasoning feels off.

File: guides/common/modules/proc_installing-standalone-hammer.adoc (hunk 5 above)
Comment (maximiliankolb) [reply guides/common/modules/proc_installing-standalone-hammer.adoc]: Yes, works for me: https://theforeman-foreman-documentation-preview-pr-3334.surge.sh/nightly/Hammer_CLI/index-foreman-deb.html#installing-standalone-hammer

**Review by maximiliankolb — APPROVED** [review]
LGTM

**Comment by asteflova** [comment]
@ekohl I'm going to keep this open for a day longer or so in case you'd like to re-review too. If you need more time that that, let me know. Otherwise, I'll merge tomorrow since I already have Max's style ack and Oleh's tech ack.

**Comment by asteflova** [comment]
Merged to "master" and cherry-picked:

9bd717ad49..7c306f5b85  3.12 -> 3.12
